lib/chef/search/query.rb

```diff
@@ -89,7 +89,7 @@ def search(type, query='*:*', *args, &block)
         if block
           response["rows"].each { |row| block.call(row) if row }
           unless (response["start"] + response["rows"].length) >= response["total"]
-            args_h[:start] = response["start"] + (args_h[:rows] || 0)
+            args_h[:start] = response["start"] + (args_h[:rows] || response["rows"].length)
```

Comment:
Well I did wonder if that wasn't a better plan... Happy to change it if you want?